src/callbacks/wandb_callbacks.py

```diff
@@ -196,3 +196,44 @@ def on_validation_epoch_end(self, trainer, pl_module):
 
             self.preds.clear()
             self.targets.clear()
+            
+
+class ImagePredictionLogger(Callback):
+    """Logs a validation batch and their predictions to wandb.
+       Example adapted from:
+       https://wandb.ai/wandb/wandb-lightning/reports/Image-Classification-using-PyTorch-Lightning--VmlldzoyODk1NzY
+    """
+
+    def __init__(self, num_samples):
+        super().__init__()
+        self.num_samples = num_samples
+
```

Comment:
This declaration is also needed in init to ensure it will work correctly even if number of validation sanity checks is set to 0 in trainer
```
self.ready = True
```